clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -362,14 +363,34 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     processNextCall();
   }
 
+  bool interrupted = false;
   {
     boost::mutex::scoped_lock lock(info->finished_mutex_);
 
+    using namespace boost::chrono;
+    system_clock::time_point now = system_clock::now();
+
     while (!info->finished_)
     {
-      info->finished_condition_.wait(lock);
+      if(timeout > 0)
+      {
+        system_clock::time_point wall_time = now + milliseconds(static_cast<int>(timeout * 1000));
+
+        if(info->finished_condition_.wait_until(lock, wall_time) == boost::cv_status::timeout)
+        {
+          ROS_ERROR("Service [%s] call failed: no response for %fsec", service_name_.c_str(), timeout);
```

Comment:
I am not sure of outputting an error in these cases is a good choice. Maybe the user uses this feature extensively and is aware that some service calls will time out but doesn't want error messages to appear in those cases?